horovod/runner/gloo_run.py

```diff
@@ -130,6 +130,11 @@ def _exec_command_fn(settings):
     :return:
     :rtype:
     """
+
+    # Non-elastic gloo runs should terminate all workers when any fail.
+    if not settings.elastic:
```

Comment:
I think the `settings.elastic` should be checked only once and `terminate_all_event` set accordingly (`None` or `threading.Event()`). Then further down only check `if terminate_all_event:` and not `settings.elastic`, because the true-clause accesses `terminate_all_event` and not `settings.elastic`.

Reply:
these seem 100% equivalent to me, but sure.

Reply:
it is logically equivalent but cleaner code